doc/manual/src/architecture/architecture.md

```diff
@@ -0,0 +1,78 @@
+# Architecture
+
+This chapter describes how Nix works.
+It should help users understand why Nix behaves as it does, and it should help developers understand how to modify Nix and how to write similar tools.
```

Comment:
>  if you want to do anything more with Nix than have a neat roll-back-able package manager, understanding the basic concepts that this document provides is critical

I thought so, too. After discussing with experienced people a lot, and doing studies with actual users, I have to disagree. We instead need better learning material, and this is why [we will put work into nix.dev](https://discourse.nixos.org/t/2022-07-07-documentation-team-meeting-notes-4/20198). I'm still convinced we have to front-load learners with specific knowledge instead of handing them inscrutable examples and templates, as Nix is different from what they usually know. But it has to be the right things, and user-oriented knowledge is different from the conceptual background presented here.

> To me, this document is about using and understanding Nix and its tooling

There should be some very condensed variant of this in the nix.dev guides. Right now it's not a priority for me. Would you be interested in giving it a try? The depth presented here should also be available on the onboarding path, but clearly optional. I'm convinced it's not really needed to use Nix effectively, but eventually studies will have to answer that question.

> If you want to start hacking on Nix itself or take inspiration for your own tools, I think there's a lot more to know.

Yes, absolutely. This architecture chapter should be the highest level of abstraction to start with. Eventually we should have paths into the source code, with additional prose sections discussing more detailed concepts, and then linking to the code to help discovering the implementation.